### ISSUE TITLE
advanced html features in languages inheriting from html

### ISSUE THREAD

**Issue by samuelcolvin** [Microsoft/vscode#49210]
With the vscode extension [Better Jinja](https://github.com/samuelcolvin/jinjahtml-vscode) the `jinja-html` language highlights jinja html templates, however some more advanced features available with the `html` langauge are not working with `jinja-html`:
* highlighting the opening tag when the cursor is in the closing tag (and visa versa) ![image](https://user-images.githubusercontent.com/4039449/39631948-95a001c0-4fab-11e8-911d-6fd571357b76.png) ![image](https://user-images.githubusercontent.com/4039449/39631931-8c49dbbe-4fab-11e8-8595-42f7ab8b1eda.png) In the second case with `jinja-html` as the language all `div`s are highlighted, not just the sibling of the tag under the cursor
* auto creating the closing tag once the opening tag is created.

And probably more, but these are the two which would be most useful.

How do I get these advanced features working in my child language definition?

I've tried copying most of the config from `extensions/html/package.json` and `extensions/html/language-configuration.json` into my extension but to no avail. I also tried looking through `extensions/html-language-features` but there's nothing obvious there to include.

This isn't as simple as adding more file extensions somewhere, even in a `.html` file if the language is changed to `jinja-html` the features above disappear.

- VSCode Version: `1.22.2 3aeede733d9a3098f7b4bdc1f66b63b0f48c1ef9 x64`
- OS Version: ubuntu 18.04

Steps to Reproduce:

1. install "Better Jinja"
2. open or create a file with some html in
3. change the language to `jinja-html`
4. try using advanced html features as described above

Does this issue occur when all extensions are disabled?: Yes (the advanced features work with the `html` language without extensions, obviously the error can only be shown with the extension installed)

**Comment by samuelcolvin** [comment]
sorry, I think this is the same as #2915. There's a lot of noise there so I'm not really sure what the status is of this but I get the impression there's not much I can do at the moment.

**Comment by aeschli** [comment]
To correctly do tag highlighting you need language smarts such as a parser for your language.
If your language syntax is pure/strict HTML (Unfortunately: most templating languages are not) then you could just the same libraries as we use for HTML or even forward all requests to HTML.

There is currently no easy way to easily add advanced HTML features to a template language. #47288 is about documenting the current state and give recommendations.

**Comment by boonpraserd** [comment]
When I used "Better Jinja", I also used "Auto Close Tag" extension to solve this problem.  Please don't forget to edit settings.json by add "jinja-html" to "auto-close-tag.activationOnLanguage".

**Comment by angelozerr** [comment]
@aeschli do you think if I create après to support my idea at https://github.com/microsoft/vscode/issues/131531#issue-978205380 could be acceptable?

I know that it doesnt cover totally this issue but it will give the capability to have a custom language which extend html language feature without extra efforts and it will work with language server written in java for instance.

**Comment by aeschli** [comment]
@angelozerr Yes, let's work on a PR for that. Can you start by opening a new one so we can discuss the approach in there?

**Comment by angelozerr** [comment]
@aeschli many thanks for your answer I created the PR at https://github.com/microsoft/vscode/pull/146731

**Comment by aeschli** [comment]
The capability you requested was implemented via the htmlLanguageParticipants contribution point, merged in commit cef02dae8d2 (July 2022, PR https://github.com/microsoft/vscode/pull/146731). Extensions can now add "htmlLanguageParticipants": [{"languageId": "jinja-html"}] to their package.json to inherit HTML language features like tag highlighting and auto-close. Closing as completed.
